### PR TITLE
pulumi-esc: init 0.9.0

### PR DESCRIPTION
**Pull request by yomaq**
## Description of changes

https://github.com/pulumi/esc
Pulumi ESC (Environments, Secrets, and Configuration)

First package, let me know if I need to make any changes.
Added package, added self as maintainer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pu/pulumi-esc/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  subPackages = "cmd/esc";
+
+  vendorHash = "sha256-Wcll/UWgCTBd1Bkbdt6kZZ6jgouWkzkFicv3CvX5hw4=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=github.com/pulumi/esc/cmd/esc/cli/version.Version=${src.rev}"
```
Comment (t0yv0) [inline pkgs/by-name/pu/pulumi-esc/package.nix:24]: Brilliant, thank you for remembering to link in the version. Is src.rev going to be just "v0.9.0" no hash added?

Comment (yomaq) [reply pkgs/by-name/pu/pulumi-esc/package.nix:24]: Yes, just comes out as "v0.9.0". Would you recommend doing it differently?

Comment (t0yv0) [reply pkgs/by-name/pu/pulumi-esc/package.nix:24]: That's great!

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Mostly looks good. I suggest using `nixfmt-rfc-style` to format per [RFC 166](https://github.com/NixOS/rfcs/pull/166). Additionally, please change the `Pulumi esc` in the title to match the `pname` (here, this is `pulumi-esc`).

**Comment by yomaq** [comment]
Done, and done. Thanks

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thanks for the PR!

**Pull request merged**